apps/application-system/api/src/app/modules/application/files/file.service.ts

```diff
@@ -71,14 +73,18 @@ export class FileService {
   ): Promise<SigningServiceResponse> {
     this.validateApplicationType(application.typeId)
     const { answers, externalData, id, state } = application as CRCApplication
-    const { nationalRegistry, parentNationalRegistry } = externalData
+    const { nationalRegistry } = externalData
     const isParentA = state === 'draft'
-
-    const parentBName = parentNationalRegistry.data.name
+    const applicant = nationalRegistry?.data
+    const selectedChildren = getSelectedChildrenFromExternalData(
+      applicant.children,
+      answers.selectChild,
+    )
+    const parentB = selectedChildren[0].otherParent
 
     switch (pdfType) {
       case PdfTypes.CHILDREN_RESIDENCE_CHANGE: {
-        const name = isParentA ? nationalRegistry.data.fullName : parentBName
+        const name = isParentA ? applicant.fullName : parentB.fullName
         const phoneNumber = isParentA
           ? answers.parentA.phoneNumber
           : answers.parentB.phoneNumber
```

Comment:
Should we join this ternary and return an object with parentinfo instead of two strings?